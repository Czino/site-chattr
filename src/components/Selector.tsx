import { ButtonHTMLAttributes } from 'react'

type Props<T> = Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'value' | 'onClick'> & {
    value: T
    selected: T
    onClick: (value: T) => void
}
export const Selector = <T,>({ value, selected, onClick, children, ...props }: Props<T>) => (
    <button
        {...props}
        className="relative p-4 w-full font-bold text-center uppercase bg-purple-50"
        onClick={() => onClick(value)}
    >
        {children}
        {value === selected && <div className="absolute right-0 bottom-0 left-0 w-full h-1 bg-highlight"></div>}
    </button>
)
