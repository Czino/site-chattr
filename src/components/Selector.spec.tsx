import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { Selector } from './Selector'

describe('Selector', () => {
    const onClick = jest.fn()
    const base = render(
        <Selector value="one" selected="two" onClick={onClick}>
            One
        </Selector>,
    ).asFragment()
    it('should render correctly', () => {
        expect(base).toMatchSnapshot()
    })
    it('should render when selected', () => {
        const { asFragment } = render(
            <Selector value="one" selected="one" onClick={onClick}>
                One
            </Selector>,
        )

        expect(asFragment()).toMatchDiffSnapshot(base)
    })
    it('should call onClick handler on click', async () => {
        render(
            <Selector value="one" selected="two" onClick={onClick}>
                One
            </Selector>,
        )
        await userEvent.click(screen.getByText('One'))
        expect(onClick).toHaveBeenCalledWith('one')
    })
})
