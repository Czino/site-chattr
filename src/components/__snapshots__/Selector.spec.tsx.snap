// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector should render correctly 1`] = `
<DocumentFragment>
  <button
    class="relative p-4 w-full font-bold text-center uppercase bg-purple-50"
  >
    One
  </button>
</DocumentFragment>
`;

exports[`Selector should render when selected 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <DocumentFragment>
    <button
      class=\\"relative p-4 w-full font-bold text-center uppercase bg-purple-50\\"
    >
      One
-     <div
-       class=\\"absolute right-0 bottom-0 left-0 w-full h-1 bg-highlight\\"
-     />
    </button>
  </DocumentFragment>"
`;
